### ISSUE TITLE
`__str__` for Layer classes.

### ISSUE THREAD

**Issue by sidps** [Lasagne/Lasagne#327]
Based on a recent discussion #324 and a previous mention #115, it might be a good idea to implement `__str__` for each of the Layer classes that provides a brief description of the layer. This would be especially useful for utils that build representations of a network, such as [draw_net](https://github.com/ebenolson/Lasagne/blob/master/examples/draw_net.py) ([discussion](https://github.com/Lasagne/Lasagne/issues/174)) and [get_network_str](https://github.com/Lasagne/Recipes/blob/master/utils/network_repr.py) ([discussion](https://github.com/Lasagne/Recipes/pull/3)).

**Comment by benanne** [comment]
Agreed! What are the precise semantics of `__str__` and `__repr__` again? Which one do we really want? Or maybe we should implement both?

**Comment by ebenolson** [comment]
`__str__` is intended to be simple and readable, `__repr__` is ideally valid python such that `eval(repr(obj)) == obj)`.  If only `__repr__` is defined, `str(obj)` will use it.

A working `__repr__` would be great, but `__str__`  is also helpful and probably simpler/less important to get perfect.

**Comment by sidps** [comment]
I'd say it's good to have both. `__repr__` seems to be used to return a description with maximal detail. Given the expected verbosity of `__repr__`, it might be good to have `__str__` return a brief alternative.

**Comment by benanne** [comment]
Sounds good, thanks for the overview :) Let's focus on `__str__` for now then.
